keras_cv/models/generative/stable_diffusion/stable_diffusion.py

```diff
@@ -0,0 +1,138 @@
+            file_hash="ad350a65cc8bc4a80c8103367e039a3329b4231c2469a1093869a345f55b1962",
+        )
+        self.text_encoder.load_weights(text_encoder_weights_fpath)
+        self.diffusion_model.load_weights(diffusion_model_weights_fpath)
+        self.decoder.load_weights(decoder_weights_fpath)
+
+    def text_to_image(
+        self,
+        prompt,
+        batch_size=1,
+        num_steps=25,
+        unconditional_guidance_scale=7.5,
+        seed=None,
+    ):
+        # Tokenize prompt (i.e. starting context)
+        inputs = self.tokenizer.encode(prompt)
+        if len(inputs) > MAX_PROMPT_LENGTH:
+            raise ValueError(
+                f"Prompt is too long (should be <= {MAX_PROMPT_LENGTH} tokens)"
+            )
+        phrase = inputs + [49407] * (MAX_PROMPT_LENGTH - len(inputs))
+        phrase = np.array(phrase)[None].astype("int32")
+        phrase = np.repeat(phrase, batch_size, axis=0)
+
+        # Encode prompt tokens + positions into a "context" vector
+        pos_ids = np.array(list(range(MAX_PROMPT_LENGTH)))[None].astype("int32")
+        pos_ids = np.repeat(pos_ids, batch_size, axis=0)
+        context = self.text_encoder.predict_on_batch([phrase, pos_ids])
+
+        # Encode unconditional tokens + positions as "unconditional context"
+        unconditional_tokens = np.array(_UNCONDITIONAL_TOKENS)[None].astype("int32")
+        unconditional_tokens = np.repeat(unconditional_tokens, batch_size, axis=0)
+        self.unconditional_tokens = tf.convert_to_tensor(unconditional_tokens)
+        unconditional_context = self.text_encoder.predict_on_batch(
+            [self.unconditional_tokens, pos_ids]
+        )
+
+        # Iterative reverse diffusion stage
+        timesteps = np.arange(1, 1000, 1000 // num_steps)
+        latent, alphas, alphas_prev = self.get_initial_parameters(
+            timesteps, batch_size, seed
+        )
+        progbar = tqdm(list(enumerate(timesteps))[::-1])
+        for index, timestep in progbar:
+            progbar.set_description(f"{index:3d} {timestep:3d}")
+            latent_prev = latent  # Set aside the previous latent vector
+            t_emb = self.get_timestep_embedding(timestep, batch_size)
+            unconditional_latent = self.diffusion_model.predict_on_batch(
+                [latent, t_emb, unconditional_context]
+            )
+            latent = self.diffusion_model.predict_on_batch([latent, t_emb, context])
+            latent = unconditional_latent + unconditional_guidance_scale * (
+                latent - unconditional_latent
+            )
+            a_t, a_prev = alphas[index], alphas_prev[index]
+            pred_x0 = (latent_prev - math.sqrt(1 - a_t) * latent) / math.sqrt(a_t)
+            latent = latent * math.sqrt(1.0 - a_prev) + math.sqrt(a_prev) * pred_x0
+
+        # Decoding stage
+        decoded = self.decoder.predict_on_batch(latent)
+        decoded = ((decoded + 1) / 2) * 255
+        return np.clip(decoded, 0, 255).astype("uint8")
```

Comment:
` tfnp.clip` ?

Reply:
It's already a numpy array (`predict_on_batch()` returns np arrays)